Build fails on Qt5 5.15.0

Build fails due to newer QNetworkReply API.

**To Reproduce**
Build with Qt5 5.15.0

**Logs**
<pre><code>/usr/include/qt/QtNetwork/qnetworkreply.h:161:10: note: declared here
  161 |     void error(QNetworkReply::NetworkError);
      |          ^~~~~
/home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.cpp: In member function ‘void multipass::LocalSocketReply::parse_status(const QByteArray&)’:
/home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.cpp:231:50: error: ‘void QNetworkReply::error(QNetworkReply::NetworkError)’ is deprecated: Use QNetworkReply::errorOccurred(QNetworkReply::NetworkError) instead [-Werror=deprecated-declarations]
  231 |         emit error(QNetworkReply::ProtocolFailure);
      |                                                  ^
In file included from /usr/include/qt/QtNetwork/QNetworkReply:1,
                 from /home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.h:23,
                 from /home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.cpp:18:
/usr/include/qt/QtNetwork/qnetworkreply.h:161:10: note: declared here
  161 |     void error(QNetworkReply::NetworkError);
      |          ^~~~~
/home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.cpp:245:30: error: ‘void QNetworkReply::error(QNetworkReply::NetworkError)’ is deprecated: Use QNetworkReply::errorOccurred(QNetworkReply::NetworkError) instead [-Werror=deprecated-declarations]
  245 |         emit error(error_code);
      |                              ^
In file included from /usr/include/qt/QtNetwork/QNetworkReply:1,
                 from /home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.h:23,
                 from /home/crow/Git/BROKEN/canonical-multipass-git/src/canonical-multipass/src/network/local_socket_reply.cpp:18:
/usr/include/qt/QtNetwork/qnetworkreply.h:161:10: note: declared here
  161 |     void error(QNetworkReply::NetworkError);
      |          ^~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [src/network/CMakeFiles/network.dir/build.make:96: src/network/CMakeFiles/network.dir/local_socket_reply.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:5111: src/network/CMakeFiles/network.dir/all] Error 2
make: *** [Makefile:161: all] Error 2</code></pre>

**Additional info**
 - OS: Arch Linux 5.4.48
- cmake version: 3.17.3

--- a/src/client/cli/cmd/mount.cpp
+++ b/src/client/cli/cmd/mount.cpp
@@ -151,7 +151,11 @@ mp::ParseCode cmd::Mount::parse_args(mp::ArgParser* parser)
 
     for (auto i = 1; i < parser->positionalArguments().count(); ++i)
     {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+        auto parsed_target = QString(parser->positionalArguments().at(i)).split(":", Qt::SkipEmptyParts);
+#else
         auto parsed_target = QString(parser->positionalArguments().at(i)).split(":", QString::SkipEmptyParts);
+#endif
 
         auto entry = request.add_target_paths();
         entry->set_instance_name(parsed_target.at(0).toStdString());

--- a/src/client/cli/cmd/set.cpp
+++ b/src/client/cli/cmd/set.cpp
@@ -123,7 +123,11 @@ mp::ParseCode cmd::Set::parse_args(mp::ArgParser* parser)
         }
         else
         {
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+            const auto keyval = args.at(0).split('=', Qt::KeepEmptyParts);
+#else
             const auto keyval = args.at(0).split('=', QString::KeepEmptyParts);
+#endif
             if (keyval.size() != 2 || keyval[0].isEmpty())
             {
                 cerr << "Bad key-value format.\n";

--- a/src/network/local_socket_reply.cpp
+++ b/src/network/local_socket_reply.cpp
@@ -147,11 +147,11 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
     auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toByteArray();
 
     // Build the HTTP method part
-    http_data += op + ' ' + request.url().path();
+    http_data += op + ' ' + request.url().path().toLatin1();
 
     if (request.url().hasQuery())
     {
-        http_data += "?" + request.url().query();
+        http_data += "?" + request.url().query().toLatin1();
     }
 
     http_data += " HTTP/1.1\r\n";

--- a/src/network/local_socket_reply.cpp
+++ b/src/network/local_socket_reply.cpp
@@ -115,7 +115,11 @@ void mp::LocalSocketReply::abort()
     close();
 
     setError(OperationCanceledError, "Operation canceled");
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    emit errorOccurred(OperationCanceledError);
+#else
     emit error(OperationCanceledError);
+#endif
 
     setFinished(true);
     emit finished();
@@ -299,7 +303,11 @@ void mp::LocalSocketReply::parse_status(const QByteArray& status)
     {
         setError(QNetworkReply::ProtocolFailure, "Malformed HTTP response from server");
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+        emit errorOccurred(QNetworkReply::ProtocolFailure);
+#else
         emit error(QNetworkReply::ProtocolFailure);
+#endif
 
         return;
     }
@@ -313,7 +321,11 @@ void mp::LocalSocketReply::parse_status(const QByteArray& status)
 
         setError(error_code, http_status_match.captured("message"));
 
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+        emit errorOccurred(error_code);
+#else
         emit error(error_code);
+#endif
     }
 }
 
@@ -323,7 +335,11 @@ bool mp::LocalSocketReply::local_socket_write(const QByteArray& data)
     if (bytes_written < 0)
     {
         setError(QNetworkReply::InternalServerError, local_socket->errorString());
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+        emit errorOccurred(QNetworkReply::InternalServerError);
+#else
         emit error(QNetworkReply::InternalServerError);
+#endif
 
         return false;
     }

--- a/src/sshfs_mount/sshfs_server.cpp
+++ b/src/sshfs_mount/sshfs_server.cpp
@@ -40,7 +40,11 @@ unordered_map<int, int> deserialise_id_map(const char* in)
 {
     unordered_map<int, int> id_map;
     QString input(in);
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 15, 0))
+    auto maps = input.split(',', Qt::SkipEmptyParts);
+#else
     auto maps = input.split(',', QString::SkipEmptyParts);
+#endif
     for (auto map : maps)
     {
         auto ids = map.split(":");

